File: acepta-el-reto/go/Bombones-de-adviento.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var numero string
	var numero1 int
	var numero2 int
	var maximo int

	for {
		_, err := fmt.Scan(&numero)
		if numero == "" || err != nil {
			break
		}
		if numero == "0" && maximo != 0 {
			fmt.Printf("%d\n", maximo)
			numero2 = 0
			maximo = 0
			continue
		}
		numero1 = numero2
		numero2, err = strconv.Atoi(numero)
		if err != nil {
			continue
		}
		if numero1+numero2 > maximo {
			maximo = numero1 + numero2
		}
	}
}

File: acepta-el-reto/Bombones-de-adviento.c
#include <stdio.h>
void main(int argc, char *argv[]) {

  long long numero1, numero2;
  long long maximo;
  while (1) {
    maximo = 0;
    numero1 = 0;
    numero2 = 0;
    scanf("%lld", &numero2);
    if (numero2 == 0) {
      break;
    }
    while (numero2 != 0) {

      if (numero1 + numero2 > maximo) {
        maximo = numero1 + numero2;
      }

      numero1 = numero2;
      scanf("%lld", &numero2);
    }
    printf("%lld\n", maximo);
  }
}
